src/main.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/parser"
+	"go/token"
+	"os"
+)
+
+type logger interface {
+	log(format string, args ...interface{}) error
+}
+
+type checker interface {
+	run(fset *token.FileSet, logger logger) error
+}
+
+type consoleLogger struct{}
+
+func (c consoleLogger) log(format string, args ...interface{}) error {
+	_, err := fmt.Printf(format, args...)
+	return err
+}
+
+var checkers = []checker{}
+
+func main() {
+	fset := token.NewFileSet()
+	dir := flag.String("dir", "", "project directory")
+	flag.Parse()
+
+	if *dir == "" {
+		flag.Usage()
+		os.Exit(1)
```

Comment:
```
		flag.Usage()
		os.Exit(1)
```
It would be helpful for user to get actual error message in this case. Like "illegal empty -dir argument".
You can factor these two lines into function with additional "message" argument.
```go
func blame(format string, args ...interface{}) {
	log.Printf(format, args...)
	flag.Usage()
	os.Exit(1)
}
```
And call it there like:
```go
blame("illegal empty -dir argument\n")
```
The `...args` would be useful, if some of `blame` calls would need formatted output.